setup.py

```diff
@@ -105,6 +108,90 @@ def run(self):
         CleanCommand.run(self)
 
 
+class BuildExtWithUpx(BuildExtCommand):
+    upx_addresses = {
+        "x86_64": "https://github.com/upx/upx/releases/download/v4.0.2/upx-4.0.2-amd64_linux.tar.xz",
```

Comment:
> why are we pinning upx to the latest release
It's the best known release

> Does this release have known issues we should be aware of?
For x86_64/aarch64 Linux, not that I'm aware.  The issues I've read for other platforms seem related to the implementation of binary objects on those platforms, or other OS features (such as execution protection).

> Should we make this version configurable?
Indeed.  We should also check the download against a static checksum to prevent man-in-the-middle attacks.  I propose doing this after #5940 has been accepted and this patch has had a chance to get through relenv.

Reply:
#5940 was merged, so I'll make this configurable as part of this PR.